processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go

```diff
@@ -38,14 +38,8 @@ func NewSpanCount(logger *zap.Logger, minSpans int32) PolicyEvaluator {
 func (c *spanCount) Evaluate(_ pcommon.TraceID, traceData *TraceData) (Decision, error) {
 	c.logger.Debug("Evaluating spans counts in filter")
 
-	traceData.Lock()
-	batches := traceData.ReceivedBatches
-	traceData.Unlock()
-
-	for _, trace := range batches {
-		if trace.SpanCount() >= int(c.minSpans) {
-			return Sampled, nil
-		}
+	if int(traceData.SpanCount.Load()) >= int(c.minSpans) {
```

Comment:
Could you please add a new test, to demonstrate the problem and ensure we don't have a regression in the future?

Reply:
Looks like 1 of the tests was already covering this but with the incorrect expected outcome. Fixed it and should now cover this logic.